### PR TITLE
Lugomateo/voi 59 deploy timething with modal

### PR DESCRIPTION
**Pull request by lugomateo**
[linear](https://linear.app/voicelayer/issue/VOI-59/deploy-timething-with-modal)

### CODE REVIEW

File: annotate.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def annotate(transcription_id):
 
     with ProgressHook() as hook:
         # load audio. https://github.com/m-bain/whisperX/issues/399
-        waveform, sample_rate = torchaudio.load(t.transcoded_file)
+        waveform, sample_rate = torchaudio.load(str(t.transcoded_file))
```
Comment (lugomateo) [inline annotate.py:79]: POSIX -> str

File: common.py [hunk 2]
```diff
@@ -52,11 +52,13 @@ class UploadInfo:
...
-            if k in inspect.signature(UploadInfo).parameters
-        })
-
+        return UploadInfo(
+            **{
+                k: v
+                for k, v in d.items()
+                if k in inspect.signature(UploadInfo).parameters
+            }
+        )
```
Comment (lugomateo) [inline common.py:61]: whoops, the formatter I'm using is putting things together quite differently, we should probably choose a common one

File: test_annotate.py [hunk 3]
```diff
@@ -17,25 +17,46 @@ class MockedStub:
...
-def test_annotate(transcription_id = "one.wav"):
+
+
+@patch("annotate.app", new=annotate_stub)
+@patch("common.app", new=annotate_stub)
+def test_annotate_one_speaker(transcription_id="one.wav"):
     with common.tmpdir_scope() as tmp:
         media_path = Path(tmp)
-        with patch('common.db', new=common.Store(media_path)):
+        with patch("common.db", new=common.Store(media_path)):
```
Comment (lugomateo) [inline test_annotate.py:28]: formatter noise...

File: test_pipeline.py [hunk 4]
```diff
@@ -81,9 +84,7 @@ def test_pipeline(transcription_id = "abc"):
             assert updates[0].state == "transcoding"
 
             assert type(updates[1]) == TranscodingProgress
-            assert updates[2].percent_done == 96
-            assert updates[5].percent_done is 100
-            assert updates[5].track.duration > 1.4
+            assert updates[4].track.duration > 1.4
```
Comment (lugomateo) [inline test_pipeline.py:87]: these assertions seem highly cpu specific. thought maybe you were running these on modal to get consistent results, but didn't see how

File: test_transcode.py [hunk 5]
```diff
@@ -48,7 +51,6 @@ def test_transcode(transcription_id = "keanu.mp3"):
             )
 
             assert len(updates) >= 3
-            assert updates[0].percent_done == 0
```
Comment (lugomateo) [inline test_transcode.py:51]: this one also seems highly cpu specific (already >20 when I run it)

Comment (purzelrakete) [reply test_transcode.py:51]: yes these are flappy

File: test_pipeline.py (hunk 4 above)
Comment (purzelrakete) [reply test_pipeline.py:87]: yep flappy

**Review by purzelrakete — APPROVED** [review]
LGTM!

**Comment by lugomateo** [comment]
I resolved 3 failing github actions tests, a few more still error out, putting those to the side for now

**Pull request merged**